go/interpreter/lfvm/memory.go

```diff
@@ -171,29 +164,33 @@ func (m *Memory) SetWord(offset uint64, value *uint256.Int, c *context) error {
 	return nil
 }
 
-func (m *Memory) Set(offset, size uint64, value []byte) error {
+// set sets the given value at the given offset.
+// Returns error if insufficient memory or offset+size overflows.
+func (m *Memory) set(offset, size uint64, value []byte) error {
 	if size > 0 {
 		if offset+size < offset {
 			return errGasUintOverflow
 		}
 		if offset+size > m.length() {
-			return fmt.Errorf("memory too small, size %d, attempted to write %d bytes at %d", m.length(), size, offset)
+			return makeInsufficientMemoryError(m.length(), size, offset)
 		}
 		copy(m.store[offset:offset+size], value)
 	}
 	return nil
 }
 
-func (m *Memory) SetWithCapacityAndGasCheck(offset, size uint64, value []byte, c *context) error {
+func makeInsufficientMemoryError(memSize, size, offset uint64) error {
+	return tosca.ConstError(fmt.Sprintf("memory too small, size %d, attempted to write %d bytes at %d", memSize, size, offset))
+}
+
+// setWithCapacityAndGasCheck sets the given value at the given offset, expands memory as needed and charges for it.
+// Returns error if insufficient gas or offset+size overflows.
+func (m *Memory) setWithCapacityAndGasCheck(offset, size uint64, value []byte, c *context) error {
```

Comment:
`WitchCapacity` and `GasCheck` is a kinda redundant. If gas is charged, there can always be enough capacity, right?

How about renaming this one to `set` as this is the safe default and rename the current `set` to `trySet` or `update` ...